test/lib/validation/ValidationError.js

```diff
@@ -0,0 +1,938 @@
+
+	const errorMessage = ValidationError.formatMessage(error);
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=required", (t) => {
+	const error = {
+		keyword: "required",
+		dataPath: "/metadata",
+		schemaPath: "../ui5.json#/definitions/metadata/required",
+		params: {missingProperty: "name"},
+		message: "should have required property 'name'"
+	};
+
+	const expectedErrorMessage = `Configuration ${chalk.underline(chalk.red("metadata"))} should have required property 'name'`;
+
+	const errorMessage = ValidationError.formatMessage(error);
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=errorMessage", (t) => {
+	const error = {
+		dataPath: "/specVersion",
+		keyword: "errorMessage",
+		message:
+`Unsupported "specVersion"
+Your UI5 CLI installation might be outdated.
+Supported specification versions: "2.0", "1.1", "1.0", "0.1"
+For details see: https://sap.github.io/ui5-tooling/pages/Configuration/#specification-versions`,
+		params: {
+			errors: [
+				{
+					dataPath: "/specVersion",
+					keyword: "enum",
+					message: "should be equal to one of the allowed values",
+					params: {
+						allowedValues: [
+							"2.0",
+							"1.1",
+							"1.0",
+							"0.1",
+						],
+					},
+					schemaPath: "#/properties/specVersion/enum",
+				},
+			],
+		},
+		schemaPath: "#/properties/specVersion/errorMessage",
+	};
+
+	const expectedErrorMessage =
+`Unsupported "specVersion"
+Your UI5 CLI installation might be outdated.
+Supported specification versions: "2.0", "1.1", "1.0", "0.1"
+For details see: https://sap.github.io/ui5-tooling/pages/Configuration/#specification-versions`;
+
+	const errorMessage = ValidationError.formatMessage(error, {});
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=additionalProperties", (t) => {
+	const error = {
+		keyword: "additionalProperties",
+		dataPath: "/resources/configuration",
+		schemaPath: "#/properties/configuration/additionalProperties",
+		params: {additionalProperty: "propertiesFileEncoding"},
+		message: "should NOT have additional properties"
+	};
+
+	const expectedErrorMessage =
+`Configuration ${chalk.underline(chalk.red("resources/configuration"))} property propertiesFileEncoding is not expected to be here`;
+
+	const errorMessage = ValidationError.formatMessage(error);
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=enum", (t) => {
+	const error = {
+		keyword: "enum",
+		dataPath: "/type",
+		schemaPath: "#/properties/type/enum",
+		params: {
+			allowedValues: ["application", "library", "theme-library", "module"]
+		},
+		message: "should be equal to one of the allowed values"
+	};
+
+	const expectedErrorMessage =
+`Configuration ${chalk.underline(chalk.red("type"))} should be equal to one of the allowed values
```

Comment:
"Configuration ${chalk.underline(chalk.red("type"))} must be equal to one of the allowed values"